### DISCUSSION TITLE
Plugin Ideas

### DISCUSSION THREAD

**Discussion by TheCompAce** [TheCompAce/Auto-GPT-Powershell#2] [Ideas]
Please post ideas for plugins that can be made.

**Comment by Kallamamran** [comment]
I'm thinking this is thought to be a "kind of offline" version of AutoChatGPT, so could this be run without API-keys and connection to the online OpenAI-model(s), but be able to do Web-searches it would be great, so...
First one sure as hell is not something noone else has thought about -> Online web search. It's a no-brainer!
The second idea, seeing how popular all these roleplaying chatbots has become, is a plugin which can connect to the pygmalion-model (or similar) for roleplay, BUT also maybe use Pinecone index for expanded memory. One thing the chatbots is lacking is memory past 2048 tokens. It would also be great if this plugin made the AI time aware so it knows how long it was since last time you chatted. This could make a impatient character which doesn't just relies to messages, but after a while maybe it asks a question itself :)
